cmd/balance-agent/main.go

```diff
@@ -570,6 +597,55 @@ func nodeDataChecker(ctx *cli.Context) error {
 	return nil
 }
 
+func fetcher(ctx *cli.Context, apiKey string) {
+	// run with --grpc-url agent-api-staging-new.bolt.observer:443 --fetch-invoices 1 --fetch-forwards 1 --fetch-payments 1
+	if ctx.String("grpc-url") != "" && apiKey != "" {
+		glog.Infof("GRPC server URL: %s", ctx.String("grpc-url"))
+
+		itf := mkGetLndAPI(ctx)()
+		if itf == nil {
+			glog.Warningf("GRPC get ligtning failure\n")
+			return
+		}
+		f, err := raw.MakeFetcher(apiKey, ctx.String("grpc-url"), itf)
+		if err != nil {
+			glog.Warningf("GRPC get fetcher failure %v\n", err)
+			return
+		}
+
+		val := int64(0)
+		var t time.Time
+
+		// 0 means turn off the fetcher, negative values means report from that exact absolute unix time (i.e., -1 means report from unix time 1), positive value
+		// try to report from that unix time (but if server says you already got something update timestamp)
+		val = ctx.Int64("fetch-invoices")
+		absVal := int64(0)
+		if val < 0 {
```

Comment:
how come this logic applies only to invoices? looks like you use invoices `val` logic for forwards and payments after line 634